### PR TITLE
:FC-899: :sparkles: creation of multiple lambda per events for Invocations by Skill

### PR DESCRIPTION
**Pull request by agalagade**

### CODE REVIEW

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 1]
```diff
@@ -34,6 +34,21 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+Please note that lambda skills can only be assigned to controller bot events, for a overview of all the supported lambda skill events head over to our [Event Sources](liveperson-functions-event-sources-overview.html).
+
+You can create more than one lambda of the same event if you set different skills for each of the lambdas. If you do not assign a skill to the lambda you are only allowed to have one lambda of the given event. 
+Some example scenarios:
+✅ Event A, Skill 1,2 + Event A, Skill 3,4
+✅ Event A, Skill 1,2 + Event A, Skill 3,4 + Event B, Default-skill
+✅ Event A, Default-skill
+❌ Event A, Skill 1,2 + Event A, Skill 3,4 +  Default-skill
+
+<img src="img/functions/faas-select-skill-2.png" alt="Functions select multiple skill" style="width:100%;"/>
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:48]: Naming does not comply

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 2]
```diff
@@ -34,6 +34,21 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+<div class="important">Changing the skills of a lambda does not require a redeployment. The change is reflected within 5 minutes after saving.</div>
```
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:50]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md (hunk 1 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:48]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 3]
```diff
@@ -34,6 +34,21 @@ You may choose a event to react to from the dropdown, the default value is `No E
 
 At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.
 
+#### Step: Lambda Skills
+To specify to which skill the lambda should react, simply select the skills during the creation/editing process of your lambda. By default, the lambda will react to all skills. If the lambda should only react to specific skills then there is a maximum of <b>50</b> skills that can be selected at any given time.
+Please note that lambda skills can only be assigned to controller bot events, for a overview of all the supported lambda skill events head over to our [Event Sources](liveperson-functions-event-sources-overview.html).
```
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:39]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 4]
```diff
@@ -34,6 +34,21 @@ You may choose a event to react to from the dropdown, the default value is `No E
 
 At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.
 
+#### Step: Lambda Skills
+To specify to which skill the lambda should react, simply select the skills during the creation/editing process of your lambda. By default, the lambda will react to all skills. If the lambda should only react to specific skills then there is a maximum of <b>50</b> skills that can be selected at any given time.
```
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:38]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:39]: resolved

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 5]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
 
 At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.
 
+### Function Skills
```
[the remark this answers is not in the document]
Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:37]: No heading is needed here, it destroys the flow.

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 6]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+Suppose you would like to limit which agent should be allowed to invoke a function during an automated event trigger. In that case, function skills functionality will help set a function to a specific skill.
+
+Select the skills during your function's creation/editing process to specify which skill the function should react to; By default, the function will respond to all skills. If the function should only respond to specific skills, then there is a maximum of **50** skills that you can select at any given time.
+
+Please note that you can only assign function skills to controller bot events; for an overview of all the supported functions skill events, head over to our [Event Sources](liveperson-functions-event-sources-overview.html#automatic-messages).
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:42]: Covered in the introduction paragraph. It makes no sense to explain to people how to achieve and then provide them with limits. 

```suggestion
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 7]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
 
 At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.
 
+### Function Skills
+Suppose you would like to limit which agent should be allowed to invoke a function during an automated event trigger. In that case, function skills functionality will help set a function to a specific skill.
+
+Select the skills during your function's creation/editing process to specify which skill the function should react to; By default, the function will respond to all skills. If the function should only respond to specific skills, then there is a maximum of **50** skills that you can select at any given time.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:40]:
```suggestion
Skills can be selected via the drop-down menu, and it is populated using the skills available in the conversational cloud. By default, the function will respond to all skills. Please be aware that the maximum number of skills is **50**.

<img src="img/functions/faas_select_skill_2.png" alt="Functions select multiple skill" style="width:100%;"/>
```

Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:40]: You might also add a short sentence explaining how the drop-down looks/behaves for other events.

Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:40]: Further, this is **only** about the creation process, please explain the updating in the other section. If needed you can add a short section, where you also mention that updates may take up to 5 minutes and are not requiring a redeployment.

Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:40]: But updating is nothing that should be discussed here.

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 8]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+Please note that you can only assign function skills to controller bot events; for an overview of all the supported functions skill events, head over to our [Event Sources](liveperson-functions-event-sources-overview.html#automatic-messages).
+
+You can create more than one function of the same event if you set different skills for each function. If you do not assign a skill to the function, you can only have one function of the given event. Some example scenarios:
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:44]:
```suggestion
You can create more than one function of the same event if you set different skills for each function. Generally, as long as they are not overlapping, you can create multiple functions for the same event. The following Scenario should highlight the implications:
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 9]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+✅ Event A, Skill 1,2 + Event A, Skill 3,4
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:46]:
```suggestion
✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4]
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 10]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+✅ Event A, Skill 1,2 + Event A, Skill 3,4 + Event B, Default-skill
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:47]: Please follow the above naming to make it more clear, as people might not get that different functions are defined between the "+"

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 11]
```diff
@@ -34,6 +34,25 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+✅ Event A, Default-skill
+❌ Event A, Skill 1,2 + Event A, Skill 3,4 + Default-skill
+
+<img src="img/functions/faas_select_skill_2.png" alt="Functions select multiple skill" style="width:100%;"/>
+
+{: .important}
+Changing the skills of a function does not require a redeployment. The change is reflected within 5 minutes after saving.
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:54]:
```suggestion
The above example, "Default-skill", refers to not selecting any skill, the default behaviour.
```

Comment (Templum) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:54]: The existing content is related to the update process, not the creation process.

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 12]
```diff
@@ -34,6 +34,32 @@ You may choose a event to react to from the dropdown, the default value is `No E
 
 At the bottom of the page you can see a toggle that will activate/deactivate the second step, that allows you to configure a [whitelist](liveperson-functions-foundations-features.html#domain-whitelisting) entry.
 
+Suppose you want to ensure that a function is triggered only for specific skills as the logic is not required for every skill. We provide you with the ability to do so. Please note that this functionality is **limited** to the [Automatic Messages](liveperson-functions-event-sources-overview.html#automatic-messages). 
+
+Skills can be selected via the drop-down menu, and it is populated using the skills available in the conversational cloud. By default, the function will respond to all skills. Please be aware that the maximum number of skills is **50**.
+
+
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:41]:
```suggestion
```

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 13]
```diff
@@ -34,6 +34,32 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+You can create more than one function of the same event if you set different skills for each function. Generally, as long as they are not overlapping, you can create multiple functions for the same event. The following Scenario should highlight the implications:
+
+✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4]
+✅  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4] + Function 3 for Event B [Default-skill]
+✅  Function 1 for Event A [Default-skill]
+❌  Function 1 for Event A [Skill 1,2] + Function 2 for Event A [Skill 3,4] + [Default-skill]
+
+<img src="img/functions/faas_select_skill.png" alt="Functions select multiple skill" style="width:100%;"/>
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:50]: The image should be above your edge case explanation. As it is not relevant to the picture.

further the image name should start with functions_*****.png like every other file

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 14]
```diff
@@ -34,6 +34,32 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+{: .notice}
+The drop-down menu is always disabled if the selected function template is not an Automatic Messages template.
+
```
Comment (Templum) [inline pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:54]: This should be actually right before or after the explanation on how to set

**Review by Templum — CHANGES_REQUESTED** [review]

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 15]
```diff
@@ -34,6 +34,32 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+To Update the skills on an existing function, you will find the skills drop-down menu inside the General information accordion menu on the right side of code Editor. Updating of skills does not require redeployment, the change is reflected within 5 minutes after saving.
+
+<img src="img/functions/faas_select_skill_edit.png" alt="Functions select multiple skill" style="width:100%;"/>
```
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md:56]: I am looking at the file name and they are starting the same way mine are functions_**.png for example. functions_ui_debugger_sidebar.png, functions_ui_debugger_output.png

[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md [hunk 16]
```diff
@@ -34,6 +34,32 @@ You may choose a event to react to from the dropdown, the default value is `No E
...
+
+{: .important}
+The above example, "Default-skill", refers to not selecting any skill, the default behaviour.
```
[the remark this answers is not in the document]
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md]: moved

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md (hunk 14 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md]: updated

File: pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md (hunk 13 above)
Comment (agalagade) [reply pages/Documents/DeveloperTools/LivePersonFunctions/GettingStarted/deep-dive-ui.md]: updated

**Review by Templum — APPROVED** [review]
LGTM

**Pull request merged**